crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,144 +1,521 @@
+    /// import A from "a";
     /// ```
     ///
-    /// Each group is independently sorted as follows:
+    /// ...is sorted as follows.
+    /// A blank line is automatically added after the header comment to ensure that the attached comment doesn't merge with the header comment.
+    ///
+    /// ```js,ignore
+    /// // Copyright notice and file header comment
     ///
-    /// ```js
-    /// import { internal } from "#internal";  // Import group 1
-    /// import sibling from "./sibling";      // Import group 1
-    /// import "z";  // Import group 2
-    /// import "a";  // Import group 3
-    /// import assert from "node:assert";  // Import group 4
-    /// import React from "react";         // Import group 4
+    /// // Attached comment for `d`
+    /// import D from "d";
+    /// // Attached comment for `e`
+    /// import E from "e";
+    /// import F from "f";
+    ///
+    /// // Detached comment (new chunk)
+    ///
+    /// // Attached comment for `a`
+    /// import A from "a";
+    /// // Attached comment for `b`
+    /// import B from "b";
     /// ```
+    ///
+    ///
+    /// ## Named imports, named exports and attributes sorting
+    ///
+    /// The sorter also sort named imports, named exports, as well as attributes.
+    /// It uses a natural sort order.
+    ///
+    /// The following code...
+    ///
+    /// ```js,expect_diagnostic
+    /// import { a, b, A, B, c10, c9 } from "a";
+    ///
+    /// export { a, b, A, B, c10, c9 } from "a";
+    ///
+    /// import special from  "special" with { "type": "ty", "metadata": "data" };
+    /// ```
+    ///
+    /// ...is sorted as follows:
+    ///
+    /// ```js,ignore
+    /// import { A, a, B, b, c9, c10 } from "a";
+    ///
+    /// export { A, a, B, b, c9, c10 } from "a";
+    ///
+    /// import special from  "special" with { "metadata": "data", "type": "ty" };
+    /// ```
+    ///
+    ///
+    /// ## Supported glob patterns
+    ///
+    /// You have to understand the structure of a source to understand which source match a glob.
+    /// A source is divided in source segments.
+    /// Every source segment is delimited by the separator `/` or the start/end of the source.
+    /// For instance `src/file.js` consists of two source segments: `src` and `file.js`.
+    ///
+    /// - star `*` that matches zero or more characters inside a source segment
+    ///
+    ///   `file.js` matches `*.js`.
+    ///   Conversely, `src/file.js` doesn't match `*.js`
+    ///
+    /// - globstar `**` that matches zero or more source segments
+    ///   `**` must be enclosed by separators `/` or the start/end of the glob.
+    ///   For example, `**a` is not a valid glob.
+    ///   Also, `**` must not be followed by another globstar.
+    ///   For example, `**/**` is not a valid glob.
+    ///
+    ///   `file.js` and `src/file.js` match `**` and `**/*.js`
+    ///   Conversely, `README.txt` doesn't match `**/*.js` because the source ends with `.txt`.
```

Comment:
```suggestion
    ///   `**` and `**/*.js` match both `file.js` and `src/file.js`.
    ///   Conversely, **/*.js` doesn't match `README.txt` because the source ends with `.txt`.
```